tests/lca/imagebasedupgrade/mgmt/negative/negative_suite_test.go

```diff
@@ -35,6 +38,48 @@ var _ = BeforeSuite(func() {
 	MGMTConfig.SeedClusterInfo = seedClusterInfo
 })
 
+var _ = AfterEach(func() {
+	By("Pull the imagebasedupgrade from the cluster")
+	ibu, err := lca.PullImageBasedUpgrade(APIClient)
+	Expect(err).NotTo(HaveOccurred(), "error pulling imagebasedupgrade resource")
+
+	if ibu.Object.Spec.Stage != "Idle" {
+		err = safeapirequest.Do(func() error {
+			ibu, err = lca.PullImageBasedUpgrade(APIClient)
+			if err != nil {
+				return err
+			}
+
+			_, err = ibu.WithStage("Idle").Update()
+			if err != nil {
+				return err
+			}
+
+			return nil
+		})
+
+		Expect(err).NotTo(HaveOccurred(), "error setting ibu to idle stage")
+
+		By("Wait until IBU has become Idle")
+		_, err = ibu.WaitUntilStageComplete("Idle")
+		Expect(err).NotTo(HaveOccurred(), "error waiting for idle stage to complete")
+	}
+
+	Eventually(func() (bool, error) {
+		ibu.Object, err = ibu.Get()
+		if err != nil {
+			return false, err
+		}
+
+		return len(ibu.Object.Status.Conditions) == 1 &&
```

Comment:
why do we check for having a single condition?

Reply:
Because when the IBU is in a proper Idle state, there is only one condition. e.g.
```
$ oc get ibu upgrade -o yaml | yq e .status -  
conditions:
  - lastTransitionTime: "2024-06-21T15:28:39Z"
    message: Idle
    observedGeneration: 197
    reason: Idle
    status: "True"
    type: Idle
observedGeneration: 197
validNextStages:
  - Prep
```